Libraries/LibWeb/Painting/DisplayListRecorder.cpp

```diff
@@ -12,13 +12,42 @@
 
 namespace Web::Painting {
 
-StackingContextTransform::StackingContextTransform(Gfx::FloatPoint origin, Gfx::FloatMatrix4x4 matrix, float scale)
+StackingContextTransform::StackingContextTransform(Gfx::FloatPoint origin, Gfx::FloatMatrix4x4 matrix, Optional<CSSPixels> parent_perspective, Gfx::FloatPoint parent_perspective_origin, float scale)
 {
     this->origin = origin.scaled(scale);
     matrix[0, 3] *= scale;
     matrix[1, 3] *= scale;
     matrix[2, 3] *= scale;
     this->matrix = matrix;
+    this->parent_perspective = parent_perspective;
+    this->parent_perspective_origin = parent_perspective_origin;
+}
+
+// https://drafts.csswg.org/css-transforms-2/#perspective-matrix
+Optional<Gfx::FloatMatrix4x4> StackingContextTransform::parent_perspective_matrix() const
+{
+    if (!parent_perspective.has_value())
+        return {};
+
+    // The perspective matrix is computed as follows:
+
+    // 1. Start with the identity matrix.
+    auto perspective_matrix = Gfx::FloatMatrix4x4::identity();
+
+    // 2. Translate by the computed X and Y values of 'perspective-origin'
+    auto computed_x = parent_perspective_origin.x();
+    auto computed_y = parent_perspective_origin.y();
+    perspective_matrix = perspective_matrix * Gfx::translation_matrix(Vector3<float>(computed_x, computed_y, 0));
```

Comment:
note: in theory you should be good with the following:

```suggestion
    // 1. Start with the identity matrix.
    // 2. Translate by the computed X and Y values of 'perspective-origin'
    auto computed_x = parent_perspective_origin.x();
    auto computed_y = parent_perspective_origin.y();
    auto perspective_matrix = Gfx::translation_matrix(Vector3<float>(computed_x, computed_y, 0));
```

Reply:
Fair point, and I've made that change now.
Though I am wondering if this isn't a thing that the compiler would optimize away anyways. Afterall, it's just a static computation.